spectrochempy/analysis/_base.py

```diff
@@ -1043,6 +1085,446 @@ def Y(self):
         return Y
 
 
+# ======================================================================================
+# Base class CrossDecompositionAnalysis
+# ======================================================================================
+
+
+class CrossDecompositionAnalysis(DecompositionAnalysis):
+    """
+    Abstract class to write analysis cross decomposition models such as `PLS`, ...
+
+    Subclass this to get a minimal structure
+
+    See Also
+    --------
+    PLSRegression : Perform a Partial Least Square Regression .
+    """
+
+    # This class is a subclass of DecompositionAnalysis, so we define only additional
+    # attributes and methods necessary for cross decomposition model.
+
+    # Get doc sections for reuse in subclass
+    _docstring.get_sections(
+        _docstring.dedent(__doc__),
+        base="CrossDecompositionAnalysis",
+        sections=["See Also"],
+    )
+
+    # ----------------------------------------------------------------------------------
+    # Public methods
+    # ----------------------------------------------------------------------------------
+
+    @_wrap_ndarray_output_to_nddataset(meta_from="_Y", title=None)
+    @_docstring.dedent
+    def predict(self, X=None, copy=True):
+        """
+        Predict targets of given observations.
+
+        Parameters
+        ----------
+        X : `NDDataset` or :term:`array-like` of shape (:term:`n_observations`\ , :term:`n_features`\ ), optional
+            New data, where :term:`n_observations` is the number of observations
+            and :term:`n_features` is the number of features.
+            if not provided, the input dataset of the `fit` method will be used.
+        copy : `bool`\ , default: `True`
+            Whether to copy X and Y, or perform in-place normalization.
```

Comment:
I do not understand why we need to set the copy parameters? We have already define a global copy parameter.

Reply:
I'll check this -- I just followed scikit-learn implementation. I will check whether both params are struictly equivalents [and actually I've not made test on the global copy parameter... So I will xheck more deeply this.

Reply:
After testing on sklearn I have the impression that `copy` parameter is actually not used, as X and Y are not changed, with copy=False.

and indeed, when looking in the [transform() code](https://github.com/scikit-learn/scikit-learn/blob/364c77e04/sklearn/cross_decomposition/_pls.py#L361), it seems that X and Y are actually not modified in place. So for the moment, I suggest removing  `copy` from the options. I will raise an issue on sklearn.